ical/tzif/tz_rule.py

```diff
@@ -200,64 +181,98 @@ def default_dst_offset(cls, values: dict[str, Any]) -> dict[str, Any]:
-    if is_julian_date:
-        tz_days = "J" + Word(nums).set_results_name("day_of_year")
-    else:
-        tz_days = month_date
-    tz_date = tz_days + Opt("/" + Group(tz_time).set_results_name("time"))
-
-    tz_rule = (
-        Group(onset).set_results_name("std")
-        + Opt(Group(onset).set_results_name("dst"))
-        + Opt(
-            ","
-            + Group(tz_date).set_results_name("dst_start")
-            + ","
-            + Group(tz_date).set_results_name("dst_end")
+    def parse(self) -> Rule:
+        """Parse the TZ string into a Rule object."""
+        std = self._parse_tz_rule_occurrence()
+        if not std:
+            raise ValueError(f"Unable to parse TZ string: {self._str}")
+
+        dst = self._parse_tz_rule_occurrence()
+        std_start = self._parse_tz_rule_date()
+        std_end = self._parse_tz_rule_date()
+
+        # either both dates are set or both are None
+        if (std_start is None) != (std_end is None):
+            raise ValueError(f"Unable to parse TZ string: {self._str}")
+
+        # make sure we have reached the end of the string
+        if self._pos < len(self._str):
+            raise ValueError(
+                f"Unable to parse TZ string: {self._str}. Unexpected end of string at position {self._pos}."
+            )
+
+        return Rule(std=std, dst=dst, dst_start=std_start, dst_end=std_end)
+
+    def _parse_tz_rule_occurrence(self) -> RuleOccurrence | None:
+        """Parse the TZ string into a RuleOccurrence object."""
+        match = _OFFSET_RE_PATTERN.match(self._str, self._pos)
+        if not match:
+            return None
+
+        self._pos = match.end()
+
+        return RuleOccurrence(
+            name=match.group("name"), offset=_parse_time(match.groupdict()) or _ZERO
         )
+
+    def _parse_tz_rule_date(self) -> Union[RuleDate, RuleDay, None]:
+        """Parse the TZ string into a RuleDate or RuleDay object."""
+        match = _START_END_RE_PATTERN.match(self._str, self._pos)
+        if not match:
+            return None
+
+        self._pos = match.end()
+
+        if match["day_of_year"] is not None:
+            return RuleDay(
+                day_of_year=int(match.group("day_of_year")),
+                time=_parse_time(match.groupdict()) or _DEFAULT_TIME_DELTA,
+            )
+
+        return RuleDate(
+            month=int(match.group("month")),
+            week_of_month=int(match.group("week_of_month")),
+            day_of_week=int(match.group("day_of_week")),
+            time=_parse_time(match.groupdict()) or _DEFAULT_TIME_DELTA,
+        )
+
+
+def build_offset_regex() -> re.Pattern[str]:
+    """Create a regular expression for the given TZ string. Alternative implementation to pyparsing."""
+    # name
+    result = r"(?P<name>(\<[+\-]?\d+\>|[a-zA-Z]+))"
+    # offset
+    result += (
+        r"((?P<hour>[+-]?\d+)(?::(?P<minutes>\d{1,2})(?::(?P<seconds>\d{1,2}))?)?)?"
+    )
+    return re.compile(result)
```

Comment:
Does a syntax like this fit?
```suggestion
    return re.compile(
        r"(?P<name>(\<[+\-]?\d+\>|[a-zA-Z]+))"   # name
        r"((?P<hour>[+-]?\d+)(?::(?P<minutes>\d{1,2})(?::(?P<seconds>\d{1,2}))?)?)?"  # offset
    )
```